nimare/meta/cbmr.py

```diff
@@ -0,0 +1,1828 @@
+        if any(
+            [
+                torch.any(torch.isnan(model.all_beta_linears[group].weight))
+                for group in self.inputs_["studies_by_group"].keys()
+            ]
+        ):
+            if self.iter == 1:  # NaN occurs in the first iteration
+                raise ValueError(
+                    """The current learing rate {str(self.lr)} gives rise to NaN values, adjust
+                    to a smaller value."""
+                )
+            all_beta_linears, all_alpha_sqrt, all_alpha = dict(), dict(), dict()
+            for group in self.inputs_["studies_by_group"].keys():
+                beta_dim = model.all_beta_linears[group].weight.shape[1]
+                beta_linear_group = torch.nn.Linear(beta_dim, 1, bias=False).double()
+                beta_linear_group.weight = torch.nn.Parameter(
+                    self.last_state["all_beta_linears." + group + ".weight"]
+                )
+                all_beta_linears[group] = beta_linear_group
+
+                if self.model == "NB":
+                    group_alpha_sqrt = torch.nn.Parameter(
+                        self.last_state["all_alpha_sqrt." + group]
+                    )
+                    all_alpha_sqrt[group] = group_alpha_sqrt
+                elif self.model == "clustered_NB":
+                    group_alpha = torch.nn.Parameter(self.last_state["all_alpha." + group])
+                    all_alpha[group] = group_alpha
+
+            model.all_beta_linears = torch.nn.ModuleDict(all_beta_linears)
+            if self.model == "NB":
+                model.all_alpha_sqrt = torch.nn.ParameterDict(all_alpha_sqrt)
+            elif self.model == "clustered_NB":
+                model.all_alpha = torch.nn.ParameterDict(all_alpha)
+
+            LGR.debug("Reset L-BFGS optimizer......")
+        else:
+            self.last_state = copy.deepcopy(
+                model.state_dict()
+            )  # need to change the variable name?
+
+        return loss
+
+    def _optimizer(self, model, lr, tol, n_iter, device):
+        """Optimize regression coefficient of CBMR via L-BFGS algorithm.
+
+        Optimization terminates if the absolute value of difference of log-likelihood in
+        two consecutive iterations is below `tol`
+
+        Parameters
+        ----------
+        model : :obj:`~nimare.dataset.Dataset`
+            Stochastic model used in CBMR.
+        lr  : :obj:`~float`
+            Learning rate of L-BFGS.
+        tol : :obj:`~float`
+            Stopping criteria of L-BFGS.
+        n_iter : :obj:`~int`
+            Maximum iterations limit of L-BFGS.
+        device : :obj:`~str`
+            Device type ('cpu' or 'cuda') represents the device on
+            which operations will be allocated.
+        """
+        optimizer = torch.optim.LBFGS(model.parameters(), lr)
+        # load dataset info to torch.tensor
+        Coef_spline_bases = torch.tensor(
```

Comment:
variable names should be all lower case (snake_case)

Reply:
Done :)